pkg/volume/cinder/attacher.go

```diff
@@ -103,27 +158,23 @@ func (attacher *cinderDiskAttacher) Attach(spec *volume.Spec, nodeName types.Nod
 	} else {
 		_, err = attacher.cinderProvider.AttachDisk(instanceID, volumeID)
 		if err == nil {
-			glog.Infof("Attach operation successful: volume %q attached to instance %q.", volumeID, instanceID)
+			if err = attacher.waitDiskAttached(instanceID, volumeID); err != nil {
+				glog.Errorf("Error waiting for volume %q to be attached from node %q: %v", volumeID, nodeName, err)
+				return "", err
+			} else {
```

Comment:
there is no need of `else` here I think, because return will anyway make the control to return from the function.